go-controller/pkg/ovn/pods.go

```diff
@@ -305,11 +305,16 @@ func (oc *DefaultNetworkController) deleteLogicalPort(pod *kapi.Pod, portInfo *l
 	// Releasing IPs needs to happen last so that we can deterministically know that if delete failed that
 	// the IP of the pod needs to be released. Otherwise we could have a completed pod failed to be removed
 	// and we dont know if the IP was released or not, and subsequently could accidentally release the IP
-	// while it is now on another pod
+	// while it is now on another pod. Releasing IPs may fail at this point if cache knows nothing about it,
+	// which is okay since node may have been deleted.
 	klog.Infof("Attempting to release IPs for pod: %s/%s, ips: %s", pod.Namespace, pod.Name,
 		util.JoinIPNetIPs(podIfAddrs, " "))
 	if err := oc.lsManager.ReleaseIPs(switchName, podIfAddrs); err != nil {
```

Comment:
I think we should return a know error in `ReleaseIPs` and ignore it similar to:
https://github.com/ovn-org/ovn-kubernetes/blob/d8976a3bda6ea8f222491faf4ba2c7b0922ee7bc/go-controller/pkg/ovn/pods.go#L271-L273